src/amo/pages/UserProfile/index.js

```diff
@@ -112,6 +113,16 @@ export class UserProfileBase extends React.Component<InternalProps> {
         }),
       );
     }
+    let idOrNameCheck = { ...location}.pathname.split('/')[4];
+    if(/^[0-9]+$/.test(idOrNameCheck)){
+      const newUrl= { ...user };
```

Comment:
You can remove that and use `user.username` in the `url:` value.

Reply:
I'm getting this error and hence, does not redirect. I just cannot use any property of 'user'.
UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'username' of undefined

Reply:
ah, I see. Yes, because we need to fetch the user first. So we need the user to send the server redirect.